keras_cv/layers/preprocessing/rand_augment.py

```diff
@@ -0,0 +1,303 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+from keras_cv import core
+from keras_cv.layers import preprocessing as cv_preprocessing
+from keras_cv.layers.preprocessing.random_augmentation_pipeline import (
+    RandomAugmentationPipeline,
+)
+from keras_cv.utils import preprocessing as preprocessing_utils
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class RandAugment(RandomAugmentationPipeline):
+    """RandAugment performs the Rand Augment operation on input images.
+
+    This layer can be thought of as an all in one image augmentation layer.  The policy
+    implemented by this layer has been benchmarked extensively and is effective on a
+    wide variety of datasets.
+
+    The policy operates as follows:
+
+    For each `layer`, the policy selects a random operation from a list of operations.
```

Comment:
"Layer" no longer refers to an actual signature argument, so the backticks are confusing -- prefer something like "For each augmentation in the range `[0, augmentations_per_image]`, "

Reply:
Can we sample in this range? Or do we need to always execute all the ops in a fixed range?

Reply:
@bhack Rate will handle this need.

Reply:
Updated docstring